source/js/components/nav/nav.jsx

```diff
@@ -26,51 +27,124 @@ export default class PrimaryNav extends React.Component {
     });
   }
 
-  toggle () {
-    if (this.state.isHidden) {
-      this.show();
+  toggleMenu () {
+    if (this.state.menuOpen) {
+      this.hideMenu();
     } else {
-      this.hide();
-    }
-  }
-
-  componentDidUpdate () {
-    if(this.props.onStateChange) {
-      this.props.onStateChange.call(this, this.state);
+      this.showMenu();
     }
   }
 
   constructor(props) {
     super(props);
 
-    this.hide = this.hide.bind(this);
+    this.hideMenu = this.hideMenu.bind(this);
+    this.showMenu = this.showMenu.bind(this);
+    this.toggleMenu = this.toggleMenu.bind(this);
+    this.escKeyPressed = this.escKeyPressed.bind(this);
 
     this.state = {
-      isHidden: true
+      menuOpen: false
     };
   }
 
-  render() {
+  escKeyPressed(e) {
+    if (e.keyCode === 27) {
+      this.hideMenu();
+    }
+  }
+
+  componentDidMount() {
+    // Close primary nav when escape is pressed
+    document.addEventListener(`keyup`, this.escKeyPressed);
+  }
+
+  componentWillUnmount() {
```

Comment:
I think so. In a general sense yes, when react does it's thing and renders a bunch of stuff, it's possible a component gets mounted and un mounted multiple times in one session, and if that happens you get leaks for the event listener. You may know this already, so what you're probably wondering is why we need it because in our situation it's very unlikely the nav ever gets unmounted. If that's what you're wondering, I figured it's better to have it anyway just in case things get copied or refactored.